src/can.rs

```diff
@@ -24,6 +24,36 @@ use core::sync::atomic::{AtomicU8, Ordering};
 const EXID_MASK: u32 = 0b1_1111_1111_1100_0000_0000_0000_0000;
 const MAX_EXTENDED_ID: u32 = 0x1FFF_FFFF;
 
+/// Options the CAN bus. This is primarily used to set bus timings, but also controls options like enabling loopback or silent mode for debugging.
+/// See  http://www.bittiming.can-wiki.info/#bxCAN for generating the timing parameters for different baud rates and clocks.
+///
+/// Use `CanBitRateOpts::default()` to get 250kbps at 32mhz system clock
+pub struct CanOpts {
+    pub brp: u16,
+    pub sjw: u8,
+    pub ts1: u8,
+    pub ts2: u8,
+    pub lbkm: bool,
+}
```

Comment:
I'm not a big fan of this approach. 

Configuring via integer values does not really guide a user of this crate towards a correct and comprehensible configuration of this peripheral, and it does lead to having many magic values in the code. 

Maybe use enums, possibly leveraging the already existing enum types provided by [stm32-rs] use the [builder-pattern](https://github.com/rust-unofficial/patterns/blob/master/patterns/builder.md)? 

[stm32-rs]: https://docs.rs/stm32f3/0.12.1/stm32f3/stm32f303/can/btr/index.html